aten/src/ATen/native/TensorFactories.h

```diff
@@ -62,9 +62,7 @@ inline void check_args(
 }
 
 inline void check_size_nonnegative(IntArrayRef size) {
-  for (auto x: size) {
-    TORCH_CHECK(x >= 0, "Trying to create tensor with negative dimension ", x, ": ", size);
-  }
+  at::check_size_nonnegative(size);
 }
```

Comment:
Can't you just say `using at::check_size_nonnegative` if you don't want to renamespace everything? (However, I kind of feel like C++ namespacing rules should make this declarations 100% unnecessary--maybe there are ADL shenanigans going on)